test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperUsageDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,316 @@
+        public static readonly string DocumentedTestAssemblyCodeBaseLocation =
+            "file:///" + DocumentedTestAssemblyLocation;
+
+        public static TheoryData CreateDescriptor_TypeData
+        {
+            get
+            {
+                var defaultLocation = DocumentedTestAssemblyLocation;
+                var defaultCodeBase = DocumentedTestAssemblyCodeBaseLocation;
+                var nonExistentLocation = defaultLocation.Replace("TestFiles", "TestFile");
+                var nonExistentCodeBase = defaultCodeBase.Replace("TestFiles", "TestFile");
+                var invalidLocation = defaultLocation + '\0';
+                var invalidCodeBase = defaultCodeBase + '\0';
+
+                // tagHelperType, expectedUsageDescriptor
+                return new TheoryData<Type, TagHelperUsageDescriptor>
+                {
+                    { CreateDocumentationTagHelperTestType(location: null, codeBase: null), null },
+                    {
+                        CreateDocumentationTagHelperTestType(defaultLocation, codeBase: null),
+                        new TagHelperUsageDescriptor(
+                            summary: "The summary for the type.",
+                            remarks: "The remarks for the type.")
+                    },
+                    {
+                        CreateDocumentationTagHelperTestType(location: null, codeBase: defaultCodeBase),
+                        new TagHelperUsageDescriptor(
+                            summary: "The summary for the type.",
+                            remarks: "The remarks for the type.")
+                    },
+                    {
+                        CreateDocumentationTagHelperTestType(defaultLocation, defaultCodeBase),
+                        new TagHelperUsageDescriptor(
+                            summary: "The summary for the type.",
+                            remarks: "The remarks for the type.")
+                    },
+                    { CreateTestType<SingleAttributeTagHelper>(defaultLocation, defaultCodeBase), null },
+                    { CreateDocumentationTagHelperTestType(nonExistentLocation, codeBase: null), null },
+                    { CreateDocumentationTagHelperTestType(location: null, codeBase: nonExistentCodeBase), null },
+                    { CreateTestType<SingleAttributeTagHelper>(invalidLocation, codeBase: null), null },
+                    { CreateDocumentationTagHelperTestType(location: null, codeBase: invalidCodeBase), null },
+                };
+            }
+        }
+
+        public static TheoryData CreateDescriptor_FallbackTypeData
+        {
+            get
+            {
+                // Faking localized dlls only to force the TagHelperUsageDescriptorFactory to search for localized XML.
+                var defaultENGBLocation = DocumentedTestAssemblyLocation.Replace(".dll", "-en-GB.dll");
+                var defaultENLocation = DocumentedTestAssemblyLocation.Replace(".dll", "-en.dll");
```

Comment:
We were spoofing the names before to avoid a hit on the XML file that's located in the non-localized folders. Now that we're checking localized folders first prior to an XML file next to the DLL i'll adjust the tests accordingly.